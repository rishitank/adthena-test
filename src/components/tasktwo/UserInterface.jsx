import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { useFetch, baseUrl } from './DataFetcher';
import ThemeDropdown from './ThemeDropdown';
import TodoInterface from './TodoInterface';

const UserInterface = ({ term }) => {
  const [query, setQuery] = useState(term);
  const themeState = useState('light');
  const [{ data: [user], error, isLoading }, setUrl] = useFetch([]);

  return (
    <div className="user-interface">
      <h2>User Interface</h2>
      <form
        onSubmit={e => {
          setUrl(`${baseUrl}/users?username=${query}`);
          e.preventDefault(); // prevent page reload
        }}
      >
        <input
          type="text"
          placeholder="Enter username"
          value={query}
          onChange={e => setQuery(e.target.value)}
          disabled={isLoading}
          required
        />
        <button type="submit">Search</button>
        <ThemeDropdown
          user={user}
          themeState={themeState}
        />
      </form>
      {error && <div className="message error">Failed to find username...</div>}
      {isLoading ? (
        <div className="message">Loading...</div>
      ) : [
        user ? (
          <div
            key="user-details"
            className={`user-details ${themeState[0]}`}
          >
            <div className="user">
              <h3><u>User Details</u></h3>
              <div>
                <strong>username: </strong>
                {user.username}
              </div>
              <div>
                <strong>email: </strong>
                {user.email}
              </div>
              <div>
                <strong>website: </strong>
                {user.website}
              </div>
            </div>
            <TodoInterface userId={user.id} />
          </div>
        ) : error !== null && (
          <div
            key="error"
            className="message error"
          >
            User not found
          </div>
        )
      ]}
    </div>
  );
};

UserInterface.defaultProps = {
  term: ''
};

UserInterface.propTypes = {
  term: PropTypes.string
};

export default UserInterface;
