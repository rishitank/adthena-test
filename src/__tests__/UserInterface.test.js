import React from 'react';
import 'whatwg-fetch';
import { render, mount } from 'enzyme';
import fetchMock from 'fetch-mock';
import UserInterface from '@components/tasktwo/UserInterface';


describe('UserInterface', () => {
  beforeAll(() => {
    global.fetch = fetch;
  });
  afterAll(() => {
    fetchMock.restore();
  });

  it('Renders correctly', () => {
    const rendered = render(<UserInterface />);
    expect(rendered).toMatchSnapshot();
  });

  it('must accept a username as input', async () => {
    const username = 'Bret';
    const wrapper = mount(<UserInterface term={username} />);
    expect(wrapper.find('input[type="text"]').instance().value).toEqual(username);

    expect(wrapper.find('.user').exists()).toBe(false);
  });
});
