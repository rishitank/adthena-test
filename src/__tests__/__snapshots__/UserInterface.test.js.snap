// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInterface Renders correctly 1`] = `
<div
  class="user-interface"
>
  <h2>
    User Interface
  </h2>
  <form>
    <input
      placeholder="Enter username"
      required=""
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Search
    </button>
    <div
      class="dropdown-holder"
    >
      <h4>
        select theme:
      </h4>
      <select
        class="dropdown"
        disabled=""
      >
        <option
          selected=""
          value="light"
        >
          light mode
        </option>
        <option
          value="dark"
        >
          dark mode
        </option>
      </select>
    </div>
  </form>
</div>
`;
